src/humanize.cr

```diff
@@ -172,10 +141,41 @@ struct Number
   # end
   #
   # humanize_length(1_420) # => "1.42 km"
-  # humanize_length(0.23)  # => "23.0 cm"
+  # humanize_length(0.23)  # => "23 cm"
   # ```
+  def humanize(precision = 3, separator = '.', delimiter = ',', *, base = 10 ** 3, significant = true) : String
+    String.build do |io|
+      humanize(io, precision, separator, delimiter, base: base, significant: significant) do |magnitude, number|
+        yield magnitude, number
+      end
+    end
+  end
+
+  # Pretty prints this number as a `String` in a human-readable format.
   #
-  # See `Int#humanize_bytes` to format a file size.
+  # This overload invokes the given *prefixes* with the order of magnitude and
+  # `self`, and expects similar return values as the overload that accepts a
+  # block.
+  #
+  # ```
+  # 123.humanize(prefixes: Number::SI_PREFIXES_PADDED) # => "123 "
+  # 0.4.humanize(prefixes: Number::SI_PREFIXES_PADDED) # => "400m"
+  # ```
+  def humanize(precision = 3, separator = '.', delimiter = ',', *, base = 10 ** 3, significant = true, prefixes : Proc) : String
+    String.build do |io|
+      humanize(io, precision, separator, delimiter, base: base, significant: significant, prefixes: prefixes)
+    end
+  end
+
+  # Pretty prints this number to the given *io* in a human-readable format.
```

Comment:
I don't like this change much because it separates the semantically equivalent overloads. Using `ditto` to copy the description of the non-streaming overload seems more helpful. The only downside is that it misses out on mentioning the `io` parameter but IMO that's acceptable.
Ideally, there should be a way to group related methods and only document the specials of each overload explicitly.

Reply:
Is this better now?

Reply:
I'm not a fan of "returns it as a `String` or writes it to the given *io*". It makes sense as a combined description for both methods, but on either single one, not so much. I would just leave the original form which doesn't make any statement about whether the result is returned or appended: "Pretty prints this number as a `String` in a human-readable format." The rest is obvious from the signature.

Reply:
And can you please swap the ditto-ed definitions back to the original order. The final diff is unnecessarily large.